google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -231,58 +235,36 @@ public CloudSdk createSdk(
         .build();
   }
 
-  private static final String CLIENT_ID_LABEL = "client_id";
-  private static final String CLIENT_SECRET_LABEL = "client_secret";
-  private static final String REFRESH_TOKEN_LABEL = "refresh_token";
-  private static final String GCLOUD_USER_TYPE_LABEL = "type";
-  private static final String GCLOUD_USER_TYPE = "authorized_user";
-
-
   @Override
   public Path stageCredentials(String googleUserName) {
-    Path credentials = doStageCredentials(googleUserName);
-    if (credentials != null) {
-      return credentials;
-    }
-
-    int addUserResult = Messages.showOkCancelDialog(
-        GctBundle.message("appengine.staging.credentials.error.message"),
-        GctBundle.message("appengine.staging.credentials.error.dialog.title"),
-        GctBundle.message("appengine.staging.credentials.error.dialog.addaccount.button"),
-        GctBundle.message("appengine.staging.credentials.error.dialog.cancel.button"),
-        Messages.getWarningIcon());
-
-    if (addUserResult == Messages.OK) {
-      Services.getLoginService().logIn();
+    if (Services.getLoginService().ensureLoggedIn(googleUserName)) {
       return doStageCredentials(googleUserName);
     }
-
     return null;
   }
 
   private Path doStageCredentials(String googleUsername) {
-    CredentialedUser projectUser = Services.getLoginService().getAllUsers()
-        .get(googleUsername);
+    Optional<CredentialedUser> projectUser =
+        Services.getLoginService().getLoggedInUser(googleUsername);
+
+    GoogleLoginState googleLoginState =
+        projectUser.map(CredentialedUser::getGoogleLoginState).orElse(null);
```

Comment:
I think a null check is in order, here, since orElse(null) won't make the method return null.